src/components/Slider/SliderBase/SliderBase.jsx

```diff
@@ -0,0 +1,90 @@
+import React, { forwardRef } from "react";
+import PropTypes from "prop-types";
+import "./SliderBase.scss";
+import { bem } from "../SliderCommons";
+import { useSliderActions, useSliderSelection, useSliderUi } from "../SliderContext";
+import { calcDimensions, getNearest, moveToPx } from "../SliderHelpers";
+import { useSliderRail } from "../SliderHooks";
+import SliderRail from "./SliderRail";
+import SliderTrack from "./SliderTrack";
+import SliderFilledTrack from "./SliderFilledTrack";
+import SliderThumb from "./SliderThumb";
+import { isArrowDownEvent, isArrowLeftEvent, isArrowRightEvent, isArrowUpEvent } from "../../../utils/dom-event-utils";
+
+function getKey(index) {
+  return index;
+}
+
+const SliderBase = forwardRef(({ className }, ref) => {
+  const { color, disabled, dragging, size, consumerBem, shapeTestId } = useSliderUi();
+  const { isRange, min, max, step, value } = useSliderSelection();
```

Comment:
I really love the separation to hooks here 😍

Reply:
Thanks